### ISSUE TITLE
Certification Warnings on startup

### ISSUE THREAD

**Issue by Tuplad** [saenzramiro/rambox#573]
Getting a massive amount of popups on startup going like:
```
The service with the following URL has an invalid authority certification.

wss://websocket.pushbullet.com/subscribe/1eGB7mlg2CbVi2y3JTGVffcnAPOjeCda

If is a Custom Service, you have to remove it and add it again, enabling the "Trust invalid authority certificates" in the Options.
```

```
The service with the following URL has an invalid authority certification.

https://api.pushbullet.com/v2/everything?modified_after=1483736875.186599

If is a Custom Service, you have to remove it and add it again, enabling the "Trust invalid authority certificates" in the Options.
```

```
The service with the following URL has an invalid authority certification.

https://api.pushbullet.com/v3/set-app-state

If is a Custom Service, you have to remove it and add it again, enabling the "Trust invalid authority certificates" in the Options.
```

Removed the service through in the /.config/Rambox folder and manually in the app.

Version: 0.4.5
Platform: Ubuntu 16.10 i686 (ia32)
Electron: 1.4.7
Chromium: 53.0.2785.143
Node: 6.5.0

EDIT: Now also with The Birdy:
```
The service with the following URL has an invalid authority certification.

https://doerllp26nx76.cloudfront.net/c/js/addpurchaseinline.js

If is a Custom Service, you have to remove it and add it again, enabling the "Trust invalid authority certificates" in the Options.
```

**Comment by Shmasya** [comment]
Same error with Skype.

**Comment by ekerazha** [comment]
Same error

**Comment by boucleinfinie** [comment]
i got this today, too, with pushbullet. Don't have it with skype, though. I have had it with openmailbox even when used as a custom service with 'trust certificates' enabled since later december.

openmailbox says this 
`The service with the following URL has an invalid authority certification.

https://stats.ombx.io/p.js

If is a Custom Service, you have to remove it and add it again, enabling the "Trust invalid authority certificates" in the Options.`

**Comment by scar45** [comment]
+1 as I am experiencing this issue as well. I remember it happening with an earlier version of Rambox, but for some reason it seems like these Certificate Warnings were triggered again for most people today. Hoping we'll get another quick Hotfix to resolve this.

**Comment by pcdog** [comment]
+1

**Comment by cyboman32** [comment]
+1

**Comment by ric2b** [comment]
+1

**Comment by YesThatAllen** [comment]
I started getting notices regarding `https://96.media.tumblr.com/cedexis/r17.gif`

the ssl certificate checks out, and I don't know which service I've had has invoked the media from tumblr (I wouldn't expect to see media from there )




-
> Rambox 0.4.5
> Electron 1.4.7
> darwin x64 16.3.0

**Comment by tbondois** [comment]
I noticed that it happens since some days with the base Zoho Email connector too... every time it's try to refresh emails

**Comment by pbirkants** [comment]
See https://github.com/saenzramiro/rambox/issues/460#issuecomment-266209291

Updating dependencies to Electron 1.4.12 and recompiling seems to have solved the warnings.

**Comment by Tuplad** [comment]
@pbirkants how did you do it ?

**Comment by hmtuan196** [comment]
+1

**Comment by edi-design** [comment]
+1 on hipchat

**Comment by replete** [comment]
+1 on hipchat

**Comment by Izin** [comment]
+1

**Comment by JimmyDaSilva** [comment]
+1 Getting much of these on Slack

**Comment by gnidorah** [comment]
+ ICQ

**Comment by Ara4Sh** [comment]
on Skype +1

**Comment by qneyrat** [comment]
👍

**Comment by Tiriel** [comment]
+1 on multiple services.

Plus, when trying to open service as custom service with Trust invalid authorities certificates checked, I get this message:

> Uncaught Exception:
TypeError: Parameter "url" must be a string, not object
    at Url.parse (url.js:87:11)
    at Object.urlParse [as parse] (url.js:81:5)
    at EventEmitter.ipcMain.on (/opt/Rambox/resources/app.asar/electron/main.js:303:45)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:191:7)
    at WebContents.<anonymous> (/opt/Rambox/resources/electron.asar/browser/api/web-contents.js:219:13)
    at emitTwo (events.js:106:13)
    at WebContents.emit (events.js:191:7)

**Comment by marcosrocha85** [comment]
+1

**Comment by lnalex** [comment]
This issue has to be classified as a critical issue. I can't use Rambox anymore, a ton of popup comes over and it's unusable.  😠

**Comment by gnidorah** [comment]
@saenzramiro For whatever reason Franz never had this issue (and I have an old version)

**Comment by dare023** [comment]
+1 Multiple services affected

**Comment by budzynowski** [comment]
+1 Mutliple services affected, from two to about eight popups, had similar issue with some old version of Franz.

**Comment by dogfuntom** [comment]
While these popups are very annoying, it seems like they're harmless. Can't you just suppress them in some way? Any hacky way will do.

**Comment by gnidorah** [comment]
> Can't you just suppress them in some way?

Should be optional setting, otherwise it will mask the real security issue if ever happen (certificate spoofing)

**Comment by Luzifer** [comment]
They should never been suppressed as that would be a huge security issue. What puzzles me is that obviously Rambox is using another root CA store than the OS / other browsers. Who is managing those and why is this the case?

**Comment by pnicorelli** [comment]
+1 for multiple services

**Comment by rokups** [comment]
@Luzifer as @pztrn pointed out in issue #460 its a chromium timebomb: https://bugs.chromium.org/p/chromium/issues/detail?id=664177

That is why simple electron update fixes issue.

**Comment by gnidorah** [comment]
@rokups wondering why Franz 4.0.4 survived this and previous certificate issue without need to update, and Rambox not...

**Comment by lnalex** [comment]
Issue solved at this time.

**Comment by rokups** [comment]
No its not. There is no new release and old one still giving certificate errors. Though everyone seems to have them different. For me riot gives errors but skype works. Others report skype broken.

**Comment by fvoordeckers** [comment]
+1 on Slack:

`The service with the following URL has an invalid authority certification. https://32.media.tumblr.com/cedexis/r17.gif?rnd=1-1-21464-0-0-17002-1813421499-_CgJqMRAUGBMiBggBENinASi7u9rgBjDmzlo4p833wwVArYWr7QRKFAgBEN8BGMF2IIqAgMAEKIWAgKAEUNSOQ1oTCAMQFBjANSDUgIDABCjHgYCgBGABahNidXR0b24xLmFtcy5odi5wcm9kggETCAMQFBjANSDUgIDABCjHgYCgBIgB1I7DGA If is a Custom Service, you have to remove it and add it again, enabling the "Trust invalid authority certificates" in the Options.`

It is Slack, so not a custom service and no "Trust invalid authority certificates" in the Options.

**Comment by cetorr** [comment]
+1 on Outlook 365 and Todoist. When can we expect a fix?

**Comment by grzegor** [comment]
Ok, there has to be done something about this kind of problems in the future — this is the second time the exact thing happens and brakes everything for me (Office 356 + Skype). I think that a simple "this might not be secure, do you really want to continue?" dialog like _all_ the major browsers do when there's a certificate problem would really be sufficient!

**Comment by DocSmirc** [comment]
+1

**Comment by justfortherec** [comment]
I have been experiencing the same issues for days. They make Rambox really hard to use. Also, it is very hard to know what is going on, because the warnings contain so little information.

Issue #589 is a proposal to turn these modal dialogs into modeless ones, in order to stop them interrupting one's work. Generally, these warnings could be much more useful, if they contained more information, as described in #575.

**Comment by Yamakaky** [comment]
Please stop posting "+1" and "same error" messages. You can react to messages instead.

**Comment by aropop** [comment]
I just build the latest commit, can confirm that it does not have the certification warnings -> it's fixed just need to build and updated i gues

**Comment by ZwS** [comment]
As temporary workaround v0.4.2 could be used.

**Comment by anpryl** [comment]
@aropop could you please share new build version?

**Comment by rokups** [comment]
Turns out we do not even need to patch anything. Simple rebuild is enough. I made linux builds if anyone wishes. https://github.com/rokups/rambox/releases/tag/0.4.5

**Comment by alex0520** [comment]
@rokups could you please explain me how to build the project, to get the deb file? I'm using rambox in ubuntu.

**Comment by rokups** [comment]
1. git clone https://github.com/saenzramiro/rambox.git
2. `cd ./rambox`
3. npm install
4. Configure env-sample.js and rename it to env.js or unpack it from app.asar from binary distribution
5. npm run sencha:compile
6. npm run build:linux64
7. Look in `./dist` dir.

I am on archlinux and build fails at some point, however it still manages to build appimage. That is what i am using. So in case build fails still look in the `./dist` dir.

**Comment by MrGKanev** [comment]
+1 on multiple services.

**Comment by philthynz** [comment]
@alex0520 you may use by deb or built source if you like https://github.com/philthynz/rambox/releases/tag/0.4.5

**Comment by Akronix** [comment]
Any chance @saenzramiro can update rambox to get this fixed for all?

**Comment by alex0520** [comment]
@philthynz I downloaded your deb, but it still shows certificate errors, when I try to use skype

**Comment by anpryl** [comment]
@rokups thank you very much. Works perfectly fine!

**Comment by Pablohn26** [comment]
Hi guys.
I am also affected by this issue. Rambox is a great software and sometimes is hard to maintain free of charge software. I have added 5$ as a [bountysource backer](https://www.bountysource.com/issues/40971858-certification-warnings-on-startup/backers) to fix this issue. Actually it is for the whole project but adding some money you can show you are seriously affected by this issue and also support this project. @saenzramiro will be glad for it.

**Comment by saenzramiro** [comment]
I'm working in version 0.5.0 that it will fix this issue.

@Pablohn26 Thank you. Of course, now I only spend time working in Rambox in my free time. I will be very happy if I can work only in Rambox, but for that the donations has to be much more. :( Thank you for your bounty.

**Comment by jas99** [comment]
+1

**Comment by Luzifer** [comment]
@saenzramiro Thank you for this tool and as a little appreciation I've also added a bounty on this…

**Comment by Izin** [comment]
@saenzramiro Thank you for Rambox. I donated for your project. Hope it will make you smile and that the version 0.5.0 will be released soon ;).

**Comment by saenzramiro** [comment]
@Luzifer @Izin Thank you for your donations.

**Comment by maibarra** [comment]
+1 @saenzramiro Muchas Gracias por tu trabajo 😄

**Comment by Jimi-James** [comment]
I'm getting this issue with Steam Chat as well.

**Comment by greyfenrir** [comment]
getting the trouble, use [skype, telegram, slack*2]

**Comment by Akronix** [comment]
Please, we all know there are still issues with this. Stop posting you're having *this* with *that*, many people get notified with each message.
Just wait for the new release. If you're really affected by this and want to have it fixed earlier, consider donate or [increase the bounty ](https://www.bountysource.com/issues/40971858-certification-warnings-on-startup/backers)

**Comment by YesThatAllen** [comment]
I've already donated, and if the fix were hard or time-consuming I'd gladly donate more.

but, the fix was published here https://github.com/saenzramiro/rambox/commit/2030ad025e528a44a89d94da4bc46d60a0690163 I just don't have the pre-requisites to build for my mac. 

What would it take to get a new set of packages made?

**Comment by rokups** [comment]
This is amazing. People are encouraged to donate to the project after active neglect of maintainer. This issue is open for a week. People are unable to use their messaging services for a week. Making a new builds for all platforms probably takes 10-15 minutes. Instead of providing 0.4.6 everyone is left with a broken software for over a week. And this is encouraged by donations. This sends a wrong message that ignorance and neglect can be rewarding. I know this will be unpopular opinion and it will be -1'ed to hell but to me this seems like holding users hostage for some donation ransom. I think this is most likely not the case but it sure looks as such. @saenzramiro could you please display some responsibility and finally provide a working build? I did compile rambox myself and i personally could not care less because i have it working, but for one person who can make his own build there is hundred who cant. Stop hurting them and your software because ignoring issues like that will definitely not allow you work full time on thing you love.

**Comment by 1beb** [comment]
I think I've clicked "ok" about 100 times in the last three days. My solution, an alternative. I use chrome generally, so I'm going to repurpose firefox with two extensions: 

Classic Theme Restorer
Tree Style Tab

The result, is rambox without rambox. Supports all services, for obvious reasons. 

![screenshot from 2017-01-22 23-51-01](https://cloud.githubusercontent.com/assets/421859/22191956/d31a77ec-e0fd-11e6-809b-203fe872cea3.png)

**Comment by jorgespiessens** [comment]
Like @ZwS said. A temporary fix is downgrading to the 0.4.2 version. Tested and works.
https://github.com/saenzramiro/rambox/releases/tag/0.4.2

**Comment by rokups** [comment]
@1beb i am stealing your idea sir! Thank you. Enough is enough..

**Comment by budzynowski** [comment]
@jorgespiessens was downgrade worked correctly with your user settings? I've got few logged on slacks, skype, gmails and i don't want to lose setting if the update is near.

**Comment by jorgespiessens** [comment]
@budzynowski Just replaced the application and everything is working fine ever since (Mac OS Sierra). No settings were lost.

**Comment by budzynowski** [comment]
@jorgespiessens thank you very much!

**Comment by gnidorah** [comment]
@1beb not a replacement at all
Rambox provides:
- unread messages tray icon
- unread messages counters in service tabs
- unified message popup notifications for all services

**Comment by rokups** [comment]
@gnidorah there is tray icon plugin for firefox. Services display unread message counts in their icons anyway (at least ones i use). They also provide desktop notifications through html5 and that translates to native desktop notifications with another plugin. So yeah it is pretty much a reasonable replacement. The only problem is window being grouped together with browser in taskbar. I am yet to figure this one out.

**Comment by gnidorah** [comment]
@rokups I believe firefox tray icon plugin will not show me that I have new messages unread.

**Comment by rokups** [comment]
Hmm yes true. It will not show certificate errors for multiple weeks as well :)

**Comment by gnidorah** [comment]
@rokups Besides Rambox, there are also Franz and probably All-In-One Messenger

**Comment by ekerazha** [comment]
@saenzramiro Instead of a 0.5.0 version, can we have the same 0.4.5 version without certification warnings? (0.4.6?) Seriously, it's **extremely** annoying.

P.S. Thank you for this awesome app.

**Comment by alex0520** [comment]
@saenzramiro why you don't try to use the alternatives that @rokups and @philthynz suggested before?

**Comment by gnidorah** [comment]
@rokups @1beb Looking at the caring some services need like https://github.com/saenzramiro/rambox/issues/413 Rambox is much a better choice!

**Comment by joshbuchea** [comment]
Just updated to v0.5.3 and all is well. Thanks @saenzramiro!

**Comment by simmac** [comment]
Wow, this "fix" is a joke. Disabling the "invalid certificate" warning is not a fix, it's a very bad workaround which enables the possibility of MITM-attacks. Please fix the problem, not the symptom! 
For example, upgrading to the newest Electron version could fix this: http://electron.atom.io/blog/2016/12/09/certificate-transparency-fix

**Comment by Akronix** [comment]
Hi. Forgive me if I'm wrong, but isn't this already fixed?

**Comment by simmac** [comment]
> Hi. Forgive me if I'm wrong, but isn't this already fixed?

Well, it looks like it has been "fixed" by disabling the warning, which means a MITM-attack will not trigger a warning anymore.

**Comment by saenzramiro** [comment]
@simmac Seems like you had time to check one commit and missed commit dfa54b957f4f6fa5d9901097ae994376f50b490a

**Comment by simmac** [comment]
@saenzramiro Sorry, I missed that one, you are correct. In this case, why did you change the default behavior (e.g this line: https://github.com/saenzramiro/rambox/commit/1643172425e5dd1c4afa1225928c9d491fa60760#diff-960075f62240ceecc56409cc2a9a143eR51)?
When I added Threema Web (https://web.threema.ch) as a custom service yesterday, I realized that the standard setting for "Trust invalid authority certificates" was `true`, which shocked me as a very security conscious person. This is the reason why I even found this issue, I searched for a reason for this behavior.

An invalid SSL certificate should never be seen as something "normal", **always** assume the worst (MITM attack with forged certificate) until proven otherwise, especially when handling very sensitive personal data like private messages.

Btw, I am sorry if my comments from yesterday seemed overly aggressive
